### PR TITLE
LPS-199071: Capital letter should not be allowed in path when creating an endpoint 

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Include validation to not allow users to set API Endpoint's path in upper case. This validation is going to be applied for singleElement and collection ones.
- Test case

**How to reproduce it?**
- Execute this curl:

```json

curl -X 'POST' \
  'http://localhost:8080/o/headless-builder/applications/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -u 'test@liferay.com:test' \
  -d '{
  "applicationStatus": "published",
  "baseURL": "app1",
  "title": "app1",
  "apiApplicationToAPIEndpoints": [
    {
      "httpMethod": "get",
      "name": "myep",
      "path": "/myEP",
      "pathParameter": "Collection"
    }
  ]
}'
```

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-199071

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java [hunk 1]
```diff
@@ -132,6 +132,9 @@ private void _addAPIApplication(
 		throws Exception {
 
 		String apiSchemaExternalReferenceCode = RandomTestUtil.randomString();
+		String path =
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java:135]: Inline this variable if it is only used once.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java [hunk 2]
```diff
@@ -94,6 +94,9 @@ private void _addAPIApplication(
 
 		String apiEndpointExternalReferenceCode = RandomTestUtil.randomString();
 		String apiSchemaExternalReferenceCode = RandomTestUtil.randomString();
+		String path =
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java:97]: Inline this variable if it is only used once.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java (hunk 2 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java:97]: SF is blaming for that

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java (hunk 1 above)
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java:135]: SF is blaming for that

Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java:135]: @4lejandrito  I did it but SF blames that it should be in a variable

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java (hunk 2 above)
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/instance/lifecycle/test/APIApplicationPublisherPortalInstanceLifecycleListenerTest.java:97]: true.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java (hunk 1 above)
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/model/listener/test/APISortRelevantObjectEntryModelListenerTest.java:135]: true.

**Review by 4lejandrito — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Comment by Hijky** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
Sent to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/142303

**Pull request closed without merging**